src/NServiceBus.AcceptanceTests/Sagas/When_saga_handles_unmapped_message.cs

```diff
@@ -0,0 +1,123 @@
+﻿namespace NServiceBus.AcceptanceTests.Sagas
+{
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_saga_handles_unmapped_message : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_throw_on_unmapped_uncorrelated_msg()
+        {
+            var id = Guid.NewGuid();
+
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<UnmappedMsgEndpoint>(b =>
+                {
+                    b.DoNotFailOnErrorMessages();
+
+                    b.When(session => session.SendLocal(new StartSagaMessage
+                    {
+                        SomeId = id
+                    }));
+                })
+                .Done(c => c.MappedEchoReceived && (c.EchoReceived || c.FailedMessages.Any()))
+                .Run();
+
+            Assert.AreEqual(true, context.StartReceived);
+            Assert.AreEqual(true, context.OutboundReceived);
+            Assert.AreEqual(true, context.MappedEchoReceived);
+            Assert.AreEqual(false, context.EchoReceived);
+            Assert.AreEqual(1, context.FailedMessages.Count);
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool StartReceived { get; set; }
+            public bool OutboundReceived { get; set; }
+            public bool EchoReceived { get; set; }
+            public bool MappedEchoReceived { get; set; }
+        }
+
+        public class UnmappedMsgEndpoint : EndpointConfigurationBuilder
+        {
+            public UnmappedMsgEndpoint()
+            {
+                EndpointSetup<DefaultServer>();
+            }
+
+            public class UnmappedMsgSaga : Saga<UnmappedMsgSagaData>,
+                IAmStartedByMessages<StartSagaMessage>,
+                IHandleMessages<MappedEchoMessage>,
+                IHandleMessages<EchoMessage>
```

Comment:
We already have a test which verifies the reply behavior

https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.AcceptanceTests/Sagas/When_an_endpoint_replies_to_a_saga.cs

I think that is enough